crates/items/src/item_stack.rs

```diff
@@ -0,0 +1,262 @@
+#![forbid(unsafe_code, warnings)]
+
+use crate::{Enchantment, Item};
+use serde::{Deserialize, Serialize};
+
+/// Represents an item stack.
+///
+/// An item stack includes an item type, an amount and a bunch of properties (enchantments, etc.)
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ItemStack {
+    /// The item type of this `ItemStack`.
+    #[serde(rename = "id")]
+    pub item: Item,
```

Comment:
I'm thinking we should make `ItemStack` fields private and only expose them through methods. The benefit is encouraging the use of safer, checked operations like `add`, `take_half`, etc. We can still expose mutable access to the fields via setters in case people need direct mutation access for whatever reason.

Reply:
Yes. That was my bad I forgot to remove the `pub` keywords. Will also add some setters. Thanks! 😄